core/shared/src/main/scala/magnolia.scala

```diff
@@ -406,7 +406,7 @@ object Magnolia {
                _root_.scala.Right(new $genericType(..${eitherVals.map(_._2)}))
              case _ =>
                _root_.scala.Left(
-                 _root_.scala.List(..${eitherVals.map(_._1)}).filter(_.isLeft).map(_.left.get)
```

Comment:
can't we simply do `_root_.scala.List(..${eitherVals.map(_._1)}).collect { case Left(v) => v }`? seems to work locally :)

Reply:
That was the first thing I tried. Unfortunately, it does not work with Scala 2.12. 
Build example: https://travis-ci.org/propensive/magnolia/jobs/621026977?utm_medium=notification&utm_source=github_status

Reply:
Ah, I didn't clean before compile locally :) you're right, it fails